Add Images tab with table to Cluster detail page

leSamo opened the pull request:
https://issues.redhat.com/browse/RHINENG-8289

Mockups: https://www.figma.com/file/LN2itBiyx8gr2aufrSib8U/%F0%9F%92%BBVulnerability-for-OpenShift?type=design&node-id=12%3A4265&mode=design&t=BfWshx27SBozXkwa-1

### What was changed:
- split page header and table store into two in Cluster detail page
- renamed cluster detail table to cluster cve table and created a second table on cluster detail page -- cluster images table along with its store
- split cluster detail table into two routes -- `/cves` and `/images` -- one for each active tab
- generalized `NoMatchingClusters` empty state to `NoMatchingItems` where parent component can choose what kind of item is displayed in the empty state
- fixed `TextFilter` to clear its text when filters are reset

### Actual results:

![Screenshot from 2024-03-30 19-47-23](https://github.com/RedHatInsights/vuln4shift-frontend/assets/8426204/16c1fc1a-979a-440d-8c94-a8ed6e93ec83)

mkholjuraev commented on src/Components/SmartComponents/ClusterDetail/ClusterImagesTable.cy.js:
```diff
@@ -0,0 +1,158 @@
+} from '../../../../cypress/utils/table';
+
+const mountComponent = () => {
+  mount(
```
Nitpick: maybe we can refactor this to separate the cypress utility function as [here](https://github.com/RedHatInsights/insights-inventory-frontend/blob/master/cypress/support/commands.js#L36). This would helps us to avoid code-duplication

leSamo replied:
I looked into this -- we could, but not that well, since some tests require `MemoryRouter`, some don't. We cannot abstract `Route` since it varies greatly -- thus without major loss in code clarity we could only abstract `<Provider store={init().getStore()}>...</Provider>` which is not worth it since we would just replace one wrapper with another wrapper.

If we were to have 4 providers as Inventory it would make sense, but not here.

mkholjuraev commented on src/Helpers/constants.js:
```diff
@@ -304,7 +324,7 @@ export const CVE_LIST_TABLE_MAPPER = (row) => ({
   key: row.id,
   cells: [
-    <Link to={'../clusters/' + row.id} key={row.id}>
+    <Link to={'../clusters/' + row.id + '/cves'} key={row.id}>
```
```suggestion
    <Link to={`../clusters/${row.id}/cves`} key={row.id}>
```

leSamo replied:
Thank you, commited.

mkholjuraev commented on src/Helpers/constants.js:
```diff
@@ -323,7 +343,7 @@ export const CLUSTER_LIST_TABLE_MAPPER = (row) => ({
   key: row.id,
   cells: [
-    <Link to={'../clusters/' + row.id} key={row.id}>
+    <Link to={'../clusters/' + row.id + '/cves'} key={row.id}>
```
```suggestion
    <Link to={`../clusters/${row.id}/cves`} key={row.id}>
```

leSamo replied:
Thank you, commited.

mkholjuraev commented on src/Helpers/miscHelper.js:
```diff
@@ -90,3 +90,8 @@ export const setupFilters = (filters, meta, defaultFilters, apply) => {
+
+export const urlChangeTab = (urlPath, newTab) => {
+  const lastSlashIndex = urlPath.lastIndexOf('/');
+  return urlPath.substring(0, lastSlashIndex + 1) + newTab;
```
should we use the react-router navigate function instead of mutating the URL pathname?

leSamo replied:
We are not directly using this function to perform navigate, we are just calculating the URL path for the `<Link> `component. I believe using `<Link>` is even clearer than using `navigate()`.

But I agree, if this function was to edit the URL it would be worse than `navigate`.

leSamo replied:
I have switched to `useNavigate` after all to save a few lines of code used to style the `Link`.

mkholjuraev commented on src/Store/Actions.js:
```diff
@@ -37,14 +37,22 @@ export const fetchCveDetails = (cveId, params) => ({
+export const fetchClusterImagesTable = (clusterId, params) => ({
+  type: ActionTypes.FETCH_CLUSTER_IMAGES_TABLE,
+  meta: {
+    timestamp: new Date(),
```
For my understanding: why do we need to pass in timestamp?

leSamo replied:
This is also discussed in Technical overview of this app:

https://docs.engineering.redhat.com/display/RHIF/Vulnerability+for+OpenShift#VulnerabilityforOpenShift-Preventingactionraceconditions

leSamo commented on src/Components/SmartComponents/ClusterDetail/ClusterDetailPage.js:
```diff
@@ -25,15 +31,44 @@ const ClusterDetailPage = () => {
-        </TextContent>
-        <ClusterDetailTable />
+        {activeTab == CLUSTER_DETAIL_TABS.cves ? (
+          <Tabs className="pf-m-light pf-v5-c-table" activeKey={0}>
```
> if these 2 tab components are here for the same tab contents (CVEs, Exposed images), I believe that there is a way to use only one of them. Am I right?

I just created a new component `TableTabs` to isolate this component, make it reusable and avoid code duplication. I wish I could also isolate the separate tabs into separate component instead of writing `renderTabs`, but Patternfly does not allow `Tab` components which are not a direct descendants of `Tabs` component.

> This is also a pretty good candidate to be shared by making it a bit more abstract and pushing to the pf-component-groups package.

This is a good thing to think about, it would depend on how the new Data view table abstraction will work. This paradigm of 1 router per tab is not followed by all Insights apps -- we would need to discuss the pros and cons of using URL path (route) vs. URL parameter to distinguish active tabs.

mkholjuraev replied:
Yup, I think that the first step is to create a shared component with the right architecture and then we can bring other applications to consistency one by one.

leSamo commented:
@mkholjuraev I have addressed all of your comments -- the main change is the creation of `TableTabs` which isolates tabs implementation from `ClusterDetailPage`

mkholjuraev reviewed (approved):
LGTM! TableTabs looks nice, thank you; tested another time, did not face any issues. 

Nitpick: when the table is empty, sorting is not disabled. This does not affect the UI at all, but other apps disable sorting.

leSamo commented:
> When the table is empty, sorting is not disabled. This does not affect the UI at all, but other apps disable sorting.
I will address this in a separate PR, thank you!
